pygsti/baseobjs/statespace.py

```diff
@@ -683,6 +683,12 @@ def __getstate__(self):
     def __setstate__(self, state_dict):
         for k, v in state_dict.items():
             self.__dict__[k] = v
+            try:
+                _ = self.__getattribute__(k)
+            except AttributeError:
+                _ = self.__dict__.pop(k)
+                self.__dict__['_' + k] = v
+                _ = self.__getattribute__(k)
```

Comment:
I remember making this change but I don't remember why. We should revert unless we have a good reason to keep as-is.